src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -241,9 +345,143 @@ def visit_doctor(
+        to_prescribe_high_sbp = high_sbp.intersection(overcome_therapeutic_inertia)
+        do_not_prescribe_high_sbp = high_sbp.difference(to_prescribe_high_sbp)
+        # If no tx prescribed, still schedule a followup
+        df.loc[do_not_prescribe_high_sbp, "to_schedule"] = 1
+        # If tx prescribed, apply ramp and schedule a followup
+        df.loc[to_prescribe_high_sbp, self.sbp_medication_column] = self.randomness.choice(
+            to_prescribe_high_sbp,
+            choices=list(
+                data_values.FIRST_PRESCRIPTION_LEVEL_PROBABILITY["sbp"]["high"].keys()
+            ),
+            p=list(data_values.FIRST_PRESCRIPTION_LEVEL_PROBABILITY["sbp"]["high"].values()),
+        )
+        df.loc[to_prescribe_high_sbp, "to_schedule"] = 1
+
+        return df
+
+    def treat_currently_medicated_sbp(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Applies the SBP treatment ramp to simulants already on medication
+        
+        Arguments:
+            df: dataframe of simulants subset to those visiting the doctor and
+                who are already on medication
+        """
+
+        # Generate indexes for simulants who overcome therapeutic inertia, are
+        # adherent to their medication, who are not already medicated, and who
+        # have low and high (measured) SBP levels
+        low_sbp = df[df["measured_level"] < data_values.SBP_THRESHOLD.HIGH].index
+        high_sbp = df[df["measured_level"] >= data_values.SBP_THRESHOLD.HIGH].index
+        overcome_therapeutic_inertia = df.index[
+            self.randomness.get_draw(df.index) > data_values.THERAPEUTIC_INERTIA_NO_START
+        ]
+        adherent = df[
+            df[self.sbp_medication_adherence_type_column]
+            == data_values.MEDICATION_ADHERENCE_TYPE.ADHERENT
+        ].index
+        not_already_max_medicated = df[
+            df[self.sbp_medication_column] < max(data_values.MEDICATION_RAMP["sbp"].values())
+        ].index
+
+        # Low-SBP patients - just schedule a followup
+        df.loc[low_sbp, "to_schedule"] = 1
+
+        # High-SBP patients
+        medication_change = (
+            high_sbp.intersection(overcome_therapeutic_inertia)
+            .intersection(adherent)
+            .intersection(not_already_max_medicated)
+        )
+        no_medication_change = high_sbp.difference(medication_change)
+        # If no tx changed, still schedule a followup
+        df.loc[no_medication_change, "to_schedule"] = 1
+        # If tx changed, move up ramp and schedule followup
+        df.loc[medication_change, self.sbp_medication_column] = (
+            df[self.sbp_medication_column] + 1
+        )
+        df.loc[medication_change, "to_schedule"] = 1
+
+        return df
+
+    def get_measurement_error(self, index, mean, sd):
+        """Return measurement error assuming normal distribution"""
+        draw = self.randomness.get_draw(index)
+        return scipy.stats.norm(loc=mean, scale=sd).ppf(draw)
```

Comment:
If you end up using this for multiple components this looks like a useful function you could throw into utilities.